cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -211,8 +266,10 @@ public void close() throws IOException {
     @Override
     public ClassLoader load(ArtifactDescriptor key) throws Exception {
       File unpackedDir = DirUtils.createTempDir(tmpDir);
-      BundleJarUtil.unpackProgramJar(key.getLocation(), unpackedDir);
-
+      String artifactFile = String.format("%s/artifacts/%s/%s", key.getArtifact().getNamespace().getId(),
```

Comment:
true, the code needs to be shared among ArtifactStore, PluginInstantiator and MapReduceRuntimeService.